docs/source/examples/sky-serve.rst

```diff
@@ -0,0 +1,333 @@
+#. **Load Balancer**: The load balancer will route the traffic to all ready replicas. It is a lightweight HTTP server that listens on the service endpoint and **HTTP-redirects** the requests to one of the replicas.
+
+All of the process group shares a single controller VM. The controller VM will be launched in the cloud with the best price/performance ratio. You can also :ref:`customize the controller resources <customizing-sky-serve-controller-resources>` based on your needs.
+
+An end-to-end LLM example
+-------------------------
+
+Below we show an end-to-end example of deploying a LLM model with Sky Serve. We'll use the `Vicuna OpenAI API Endpoint YAML <https://github.com/skypilot-org/skypilot/blob/master/llm/vicuna/serve-openai-api-endpoint.yaml>`_ as an example:
+
+.. code-block:: yaml
+
+    resources:
+      ports: 8080
+      accelerators: A100:1
+      disk_size: 1024
+      disk_tier: high
+
+    setup: |
+      conda activate chatbot
+      if [ $? -ne 0 ]; then
+        conda create -n chatbot python=3.9 -y
+        conda activate chatbot
+      fi
+
+      # Install dependencies
+      pip install "fschat[model_worker,webui]==0.2.24"
+      pip install protobuf
+
+    run: |
+      conda activate chatbot
+
+      echo 'Starting controller...'
+      python -u -m fastchat.serve.controller > ~/controller.log 2>&1 &
+      sleep 10
+      echo 'Starting model worker...'
+      python -u -m fastchat.serve.model_worker \
+                --model-path lmsys/vicuna-${MODEL_SIZE}b-v1.3 2>&1 \
+                | tee model_worker.log &
+
+      echo 'Waiting for model worker to start...'
+      while ! `cat model_worker.log | grep -q 'Uvicorn running on'`; do sleep 1; done
+
+      echo 'Starting openai api server...'
+      python -u -m fastchat.serve.openai_api_server --host 0.0.0.0 --port 8080 | tee ~/openai_api_server.log
+
+    envs:
+      MODEL_SIZE: 7
+
+The above SkyPilot Task YAML will launch an OpenAI API endpoint with a 7B Vicuna model. This YAML can be used with :code:`sky launch` to launch a single replica of the service. By adding a :code:`service` section to the YAML, we can scale it into multiple replicas across multiple regions/clouds:
+
+.. code-block:: yaml
+
+    # vicuna.yaml
+    service:
+      readiness_probe: /v1/models
+      replicas: 3
+
+    resources:
+      ports: 8080
+      # Here goes other resources config
+
+    # Here goes other task config
+
+Now you have a Service YAML that can be used with Sky Serve! Simply run :code:`sky serve up vicuna.yaml -n vicuna` to deploy the service (use :code:`-n` to give your service a name!). After a while, there will be an OpenAI Compatible API endpoint ready to accept traffic (:code:`44.201.113.28:30001` in the following example):
+
+.. image:: ../images/sky-serve-status-vicuna-ready.png
+    :width: 800
+    :align: center
+    :alt: sky-serve-status-vicuna-ready
+
```

Comment:
We should add a CURL based command like the user doc. Does something like this work
```
Then it’s ready to accept traffic!

$ curl -L Y.Y.Y.Y:8082/generate \
    -X POST \
    -d '{"inputs":"What is Deep Learning?","parameters":{"max_new_tokens":20}}' \
    -H 'Content-Type: application/json'
{"generated_text":"\n nobody knows"}
```